src/Filters/Snails.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Matecat\SubFiltering\Filters;
+
+use Matecat\SubFiltering\Commons\AbstractHandler;
+use Matecat\SubFiltering\Commons\Constants;
+
+class Snails extends AbstractHandler
+{
+    /**
+     * @inheritDoc
+     */
+    public function transform( $segment )
+    {
+        preg_match_all( '/@@[^<>\s%]+?@@|(?<!{)@[^<>\s@]+?@/', $segment, $html, PREG_SET_ORDER );
```

Comment:
The negative lookbehind is not needed here. You can use simple assertion:
`/@@[^<>\s]+?@@|@[^<>\s@]+?@/`

Reply:
The regex was updated as requested